plugins/rest/rest_handlers.go

```diff
@@ -30,174 +30,192 @@ import (
+			}
 		}
-		defer ch.Close()
+		return ifs, err
+	}); err != nil {
+		return err
+	}
 
-		fibHandler, err := l2plugin.NewFibVppHandler(ch, nil, make(chan *l2plugin.FibLogicalReq), plugin.Log, nil)
-		if err != nil {
-			plugin.Log.Errorf("Error creating VPP handler: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		res, err := fibHandler.DumpFIBTableEntries()
-		if err != nil {
-			plugin.Log.Errorf("Error: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, nil)
-			return
-		}
+	return nil
+}
 
-		plugin.Log.Debug(res)
-		formatter.JSON(w, http.StatusOK, res)
+func (plugin *Plugin) registerBfdHandlers() error {
+	// GET BFD configuration
+	if err := plugin.registerHTTPHandler(bfd.RestBfdKey(), GET, func() (interface{}, error) {
+		return plugin.bfdHandler.DumpBfdSingleHop()
+	}); err != nil {
+		return err
 	}
+	// GET BFD sessions
+	if err := plugin.registerHTTPHandler(bfd.RestSessionKey(), GET, func() (interface{}, error) {
+		return plugin.bfdHandler.DumpBfdSessions()
+	}); err != nil {
+		return err
+	}
+	// GET BFD authentication keys
+	if err := plugin.registerHTTPHandler(bfd.RestAuthKeysKey(), GET, func() (interface{}, error) {
+		return plugin.bfdHandler.DumpBfdAuthKeys()
+	}); err != nil {
+		return err
+	}
+
+	return nil
 }
 
-// xconnectPairsGetHandler - used to get list of all connect pairs (transmit and receive interfaces)
-func (plugin *Plugin) xconnectPairsGetHandler(formatter *render.Render) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
+// Registers L2 plugin REST handlers
+func (plugin *Plugin) registerL2Handlers() error {
+	// GET bridge domain IDs
+	if err := plugin.registerHTTPHandler(l2.RestBridgeDomainIDKey(), GET, func() (interface{}, error) {
+		return plugin.bdHandler.DumpBridgeDomainIDs()
+	}); err != nil {
+		return err
+	}
+	// GET bridge domains
+	if err := plugin.registerHTTPHandler(l2.RestBridgeDomainKey(), GET, func() (interface{}, error) {
+		return plugin.bdHandler.DumpBridgeDomains()
+	}); err != nil {
+		return err
+	}
+	// GET FIB entries
+	if err := plugin.registerHTTPHandler(l2.RestFibKey(), GET, func() (interface{}, error) {
+		return plugin.fibHandler.DumpFIBTableEntries()
+	}); err != nil {
+		return err
+	}
+	// GET cross connects
+	if err := plugin.registerHTTPHandler(l2.RestXConnectKey(), GET, func() (interface{}, error) {
+		return plugin.xcHandler.DumpXConnectPairs()
+	}); err != nil {
+		return err
+	}
 
-		plugin.Log.Debug("Getting list of all xconnect pairs")
+	return nil
+}
 
-		// create an API channel
-		ch, err := plugin.GoVppmux.NewAPIChannel()
-		if err != nil {
-			plugin.Log.Errorf("Error creating channel: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		defer ch.Close()
+// registerHTTPHandler is common register method for all handlers
+func (plugin *Plugin) registerHTTPHandler(key, method string, f func() (interface{}, error)) error {
+	var err error
```

Comment:
This error is not used anywhere.. why does this function need to return error anyway? The `HTTPHandlers.RegisterHTTPHandler` doesn't return one.